lib/experimental/fonts/font-library/class-wp-font-collection.php

```diff
@@ -115,33 +115,45 @@ public function __construct( $config ) {
 	 *  }
 	 * @return bool True if the font collection config is valid and false otherwise.
 	 */
-	public static function is_config_valid( $config ) {
+	public static function is_config_valid( $slug, $config ) {
+		if ( empty( $slug ) || ! is_string( $slug ) ) {
+			_doing_it_wrong( __METHOD__, __( 'Font Collection slug is required as a non-empty string.', 'gutenberg' ), '6.5.0' );
+			return false;
+		}
+
 		if ( empty( $config ) || ! is_array( $config ) ) {
 			_doing_it_wrong( __METHOD__, __( 'Font Collection config options are required as a non-empty array.', 'gutenberg' ), '6.5.0' );
 			return false;
 		}
 
-		$required_keys = array( 'slug', 'name' );
-		foreach ( $required_keys as $key ) {
-			if ( empty( $config[ $key ] ) ) {
-				_doing_it_wrong(
-					__METHOD__,
-					// translators: %s: Font collection config key.
-					sprintf( __( 'Font Collection config %s is required as a non-empty string.', 'gutenberg' ), $key ),
-					'6.5.0'
-				);
-				return false;
-			}
+		if ( empty( $config['name'] ) || ! is_string( $config['name'] ) ) {
+			_doing_it_wrong( __METHOD__, __( 'Font Collection name is required as a non-empty string.', 'gutenberg' ), '6.5.0' );
+			return false;
 		}
 
 		if (
-			( empty( $config['src'] ) && empty( $config['font_families'] ) ) ||
-			( ! empty( $config['src'] ) && ! empty( $config['font_families'] ) )
+			( isset( $config['src'] ) && isset( $config['font_families'] ) ) ||
+			( ! isset( $config['src'] ) && ! isset( $config['font_families'] ) )
 		) {
 			_doing_it_wrong( __METHOD__, __( 'Font Collection config "src" option OR "font_families" option are required.', 'gutenberg' ), '6.5.0' );
 			return false;
 		}
 
+		if ( isset( $config['src'] ) && ! is_string( $config['src'] ) ) {
+			_doing_it_wrong( __METHOD__, __( 'Font Collection config "src" option is required as a non-empty string.', 'gutenberg' ), '6.5.0' );
+			return false;
+		}
+
+		if ( isset( $config['font_families'] ) && ( empty( $config['font_families'] ) || ! is_array( $config['font_families'] ) ) ) {
+			_doing_it_wrong( __METHOD__, __( 'Font Collection config "font_families" option is required as a non-empty array.', 'gutenberg' ), '6.5.0' );
+			return false;
+		}
+
+		if ( isset( $config['categories'] ) && ( empty( $config['categories'] ) || ! is_array( $config['categories'] ) ) ) {
+			_doing_it_wrong( __METHOD__, __( 'Font Collection config "categories" option is required as a non-empty array.', 'gutenberg' ), '6.5.0' );
+			return false;
+		}
```

Comment:
Should these three `_doing_it_wrong` calls use `sprintf` placeholders, similar to [this change](https://github.com/WordPress/gutenberg/commit/18f996342b4ba37542eaa87a727a409997bd162f)?

Reply:
:+1: Added that here: https://github.com/WordPress/gutenberg/pull/58269/commits/b826ceadc2a2571a83d9d99093ed52913493ae59